src/lib/crypto/kimchi_bindings/stubs/src/pasta_fp_plonk_proof.rs

```diff
@@ -312,7 +312,70 @@ pub fn caml_pasta_fp_plonk_proof_example_with_foreign_field_mul(
     let proof = ProverProof::create_recursive::<EFqSponge, EFrSponge>(
         &group_map,
         witness,
-        &vec![],
+        &[],
+        &index,
+        vec![],
+        None,
+    )
+    .unwrap();
+    (CamlPastaFpPlonkIndex(Box::new(index)), proof.into())
+}
+
+#[ocaml_gen::func]
+#[ocaml::func]
+pub fn caml_pasta_fp_plonk_proof_example_with_range_check(
+    srs: CamlFpSrs,
+) -> (CamlPastaFpPlonkIndex, CamlProverProof<CamlGVesta, CamlFp>) {
+    use ark_ff::Zero;
+    use commitment_dlog::srs::{endos, SRS};
+    use kimchi::circuits::{
+        constraints::ConstraintSystem, gate::CircuitGate, polynomials::range_check, wires::Wire,
+    };
+    use num_bigint::BigUint;
+    use num_bigint::RandBigInt;
+    use o1_utils::{foreign_field::BigUintForeignFieldHelpers, BigUintFieldHelpers};
+    use rand::{rngs::StdRng, SeedableRng};
+
+    let rng = &mut StdRng::from_seed([255u8; 32]);
+
+    // Create range-check gadget
+    let (mut next_row, mut gates) = CircuitGate::<Fp>::create_multi_range_check(0);
+
+    // Create witness
+    let witness = range_check::witness::create_multi::<Fp>(
+        rng.gen_biguint_range(&BigUint::zero(), &BigUint::two_to_limb())
+            .to_field()
+            .expect("failed to convert to field"),
+        rng.gen_biguint_range(&BigUint::zero(), &BigUint::two_to_limb())
+            .to_field()
+            .expect("failed to convert to field"),
+        rng.gen_biguint_range(&BigUint::zero(), &BigUint::two_to_limb())
+            .to_field()
+            .expect("failed to convert to field"),
+    );
+
+    // Temporary workaround for lookup-table/domain-size issue
+    for _ in 0..(1 << 13) {
+        gates.push(CircuitGate::zero(Wire::for_row(next_row)));
```

Comment:
I think we should have a circuit gate creator that simply extends it to the last row of the circuit, no need to tell the row. Meaning, that it automatically takes gates.len() and puts it there. But anyway, this is independent of the PR

Reply:
We already have that for range-check. 👍🏼

Reply:
Good idea.